Extension/src/LanguageServer/protocolFilter.ts

```diff
@@ -7,6 +7,29 @@
 import { Middleware } from 'vscode-languageclient';
 import { ClientCollection } from './clientCollection';
 import { Client } from './client';
+import { SourceFileConfiguration, CustomConfigurationProvider } from '../api';
+import { customConfigurationProviders } from './extension';
+
+function runThenableWithTimeout(promise: Thenable<any>, ms: number): Thenable<any> {
+    let timer: NodeJS.Timer;
+
+    // Create a promise that rejects in <ms> milliseconds
+    let timeout: Promise<any> = new Promise((resolve, reject) => {
+        timer = setTimeout(() => {
+            clearTimeout(timer);
+            reject("Timed out in " + ms + "ms.");
+        }, ms);
+    });
+
+    // Returns a race between our timeout and the passed in promise
+    return Promise.race([
+        promise,
```

Comment:
One other thing about this race is that your extension won't be notified that we're going to ignore your result, but at the same time, the request isn't cancelled.  Should we attempt to cancel it?